app/src/main/cpp/ControllerContainer.cpp

```diff
@@ -203,38 +204,42 @@ void ControllerContainer::SetHandJointLocations(const int32_t aControllerIndex,
     if (!m.root)
         return;
 
-    // Initialize hand joints if needed
-    if (controller.handJointTransforms.size() == 0) {
-        controller.handJointTransforms.resize(jointTransforms.size());
+    CreationContextPtr create = m.context.lock();
 
-        CreationContextPtr create = m.context.lock();
-
-        // Initialize left hand action button, which for now triggers back navigation.
-        // Note that Quest's runtime already shows the hamburger menu button when left
-        // hand is facing head.
+    // Initialize left hand action button, which for now triggers back navigation.
+    // Note that Quest's runtime already shows the hamburger menu button when left
+    // hand is facing head.
 #if !defined(OCULUSVR)
-        if (controller.leftHanded) {
-            TextureGLPtr texture = create->LoadTexture("menu.png");
-            assert(texture);
-            texture->SetTextureParameter(GL_TEXTURE_MAG_FILTER, GL_LINEAR);
-            texture->SetTextureParameter(GL_TEXTURE_MIN_FILTER, GL_LINEAR);
-
-            const float iconWidth = 0.03f;
-            const float aspect = (float)texture->GetWidth() / (float)texture->GetHeight();
-
-            QuadPtr icon = Quad::Create(create, iconWidth, iconWidth / aspect, nullptr);
-            icon->SetTexture(texture, texture->GetWidth(), texture->GetHeight());
-            icon->UpdateProgram("");
-
-            controller.handActionButtonToggle = Toggle::Create(create);
-            controller.handActionButtonTransform = Transform::Create(create);
-            controller.handActionButtonToggle->AddNode(controller.handActionButtonTransform);
-            controller.handActionButtonTransform->AddNode(icon->GetRoot());
-            controller.handActionButtonToggle->ToggleAll(false);
-            m.root->AddNode(controller.handActionButtonToggle);
-        }
+    if (controller.leftHanded && controller.handActionButtonToggle == nullptr) {
```

Comment:
Let's keep the (wrong) indentation as it was in order not to pollute the already big change and unnecessarily alter the blame history of those lines

Reply:
Unfortunately it is not really a wrong indentation, this block was just taken out of an outer if block that is no longer relevant.

Previously, we were using the lazy construction of the hand joint spheres to also construct the back-nav icon. But now that the spheres are only constructed for Pico, and the back-nav needs to be constructed for all devices except Oculus, we had to rely on other mechanism to perform the lazy-load.

This part of the code is the most confusing one, partly because we had to keep the hand joint spheres code path but only for Pico, and also because we had mingled the back-nav code inside the wrong block :/